Sources/CNIOAtomics/src/c-fastatomics.c

```diff
@@ -0,0 +1,75 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#include <stdlib.h>
+#include <stdatomic.h>
+#include <stdbool.h>
+#include <inttypes.h>
+#include <stdio.h>
+
+#include "../include/CNIOAtomics.h"
+#include "cpp_magic.h"
+
+#define MAKE(type) /*
+*/ void catmc_fast_atomic_##type##_create_with_existing_storage(struct catmc_fast_atomic_##type *storage, type value) { /*
```

Comment:
@2bjake instead of duplicating everything here, couldn’t we just declare the existing structs in the header like you do for the fast atomics? There’s no actual need to only forward declare them. I’m sure we can get away with just one copy of the C stuff

Reply:
I originally tried this but declaring the structs in the header causes the generated Swift interface of these functions to change. This means the ```AtomicPrimitive``` protocol would have to change as well (compare ```AtomicPrimitive``` and ```FastAtomicPrimitive```). Wouldn't this result in a backward incompatible API change?

Reply:
Wait, I see a way where the pointer type of ```AtomicPrimitive``` wouldn't have to change pointer types by moving away from pointfree conformance and wrapping the c functions. Something like: 
```swift
public static let opaque_add: (OpaquePointer, Self) -> Self = { catmc_fast_atomic__Bool_add(UnsafeMutablePointer<AtomicWrapper>($0), $1) }
```

But there will still need to be a ```FastAtomicPrimitive``` protocol, right? So I can add the create_with_storage method and the AtomicWrapper associatedtype in a non-breaking way.

Reply:
OK, I was able to remove the duplicate c code. I was also able to dramatically reduce the boilerplate for ```AtomicPrimitive``` conformance by providing default implementation computed properties which rely on ```FastAtomicPrimitive```!